solutions/ac-IndependentSet.mdx

```diff
@@ -78,3 +82,74 @@ int main() {
 	return 0;
 }
 ```
+
+</CPPSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
+    public static ArrayList<Integer>[] adj;
+    public static boolean[] visited;
+    public static long[][] dp;
+    public static int MOD = (int) 1e9 + 7;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
+        int N = Integer.parseInt(r.readLine());
+        visited = new boolean[N];
+        adj = new ArrayList[N];
+        dp = new long[2][N];
+
+        for (int i = 0; i < N; i++) {
+            adj[i] = new ArrayList<>();
+        }
+
+        // Read in the edges.
```

Comment:
lines 110, 111, and 121 are unnecessary imo

Reply:
i think more space is good, but "Read in the edges." is probably unnecessary

Reply:
spaces are used to separate lines of code into their own "categories" i think dividing those groups of code was kinda unnecessary

Reply:
well each loop is technically its own category... and there are spaces around all the loops so it looks nice and consistent imo

but i think line 125 is unnecessary

Reply:
110 is unnecessary because it implies that creating the adjacency list is somehow very separate from reading in the edges, which i feel is kinda wrong
but you have a point